osu.Game/Overlays/SkinEditor/SkinComponentToolbox.cs

```diff
@@ -23,14 +24,22 @@ public partial class SkinComponentToolbox : EditorSidebarSection
     {
         public Action<Type>? RequestPlacement;
 
-        private readonly SkinComponentsContainer? target;
+        private readonly SkinComponentsContainer target;
+
+        private readonly RulesetInfo? ruleset;
 
         private FillFlowContainer fill = null!;
 
-        public SkinComponentToolbox(SkinComponentsContainer? target = null)
-            : base(target?.Lookup.Ruleset == null ? SkinEditorStrings.Components : LocalisableString.Interpolate($"{SkinEditorStrings.Components} ({target.Lookup.Ruleset.Name})"))
+        /// <summary>
+        /// Create a new component toolbox for the specified taget.
+        /// </summary>
+        /// <param name="target">The target. This is mainly used as a dependency source to find candidate components.</param>
+        /// <param name="ruleset">A ruleset to filter components by. If null, only components which are not ruleset-specific will be included.</param>
+        public SkinComponentToolbox(SkinComponentsContainer target, RulesetInfo? ruleset)
+            : base(ruleset == null ? SkinEditorStrings.Components : LocalisableString.Interpolate($"{SkinEditorStrings.Components} ({ruleset.Name})"))
```

Comment:
Unfortunatly, this breaks the test at `TestSceneSkinEditorComponentsList.TestToggleEditor()`

Reply:
Fixed in aca8310cd1df87a131e1e1dfc7d0bc047101c578.